### ISSUE TITLE
Build: compiling dependency `electrs-librocksdb-sys` fails with gcc 13

### ISSUE THREAD

**Issue by antonilol** [romanz/electrs#877]
**Have you read the documentation?**
Yes.

**Did you double-check that you installed all dependencies?**
Yes.

**Which command failed?**
`cargo build`

**What was the error message?**

<details>
<summary>Error message</summary>

```
error occurred: Command "c++" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-4" "-fno-omit-frame-pointer" "-m64" "-I" "rocksdb/include/" "-I" "rocksdb/" "-I" "rocksdb/third-party/gtest-1.8.1/fused-src/" "-I" "snappy/" "-I" "/home/antoni/.cargo/registry/src/github.com-1ecc6299db9ec823/zstd-sys-2.0.8+zstd.1.5.5/zstd/lib" "-I" "." "-Wall" "-Wextra" "-std=c++17" "-Wsign-compare" "-Wshadow" "-Wno-unused-parameter" "-Wno-unused-variable" "-Woverloaded-virtual" "-Wnon-virtual-dtor" "-Wno-missing-field-initializers" "-Wno-strict-aliasing" "-Wno-invalid-offsetof" "-msse2" "-std=c++17" "-DSNAPPY=1" "-DZSTD=1" "-DNDEBUG=1" "-DOS_LINUX" "-DROCKSDB_PLATFORM_POSIX" "-DROCKSDB_LIB_IO_POSIX" "-DROCKSDB_SUPPORT_THREAD_LOCAL" "-DHAVE_UINT128_EXTENSION=1" "-o" "/home/antoni/github/electrs/target/debug/build/electrs-librocksdb-sys-09eda1a4f00201a3/out/rocksdb/table/block_based/data_block_hash_index.o" "-c" "rocksdb/table/block_based/data_block_hash_index.cc" with args "c++" did not execute successfully (status code exit status: 1).
```

</details>

**System**
OS name and version: archlinux 6.1.27-1-lts with gcc 13.1.1-1
rustc version: rustc 1.63.0 (4b91a6ea7 2022-08-08)
cargo version: cargo 1.63.0 (fd9c4297c 2022-07-01)

**Compilation**
Linking: static
Cross compilation: no
Target architecture: x86_64

**Additional context**

gcc recently got updated on archlinux from 12 where it used to work to 13 where it no longer builds. downgrading gcc (not recommended) or installing an older version (like gcc11 from the aur) and building with `CXX='/bin/c++-11' cargo build` will work around this

**Comment by romanz** [comment]
Many thanks for reporting this issue!
BTW, does building latest [rust-rocksdb](https://github.com/rust-rocksdb/rust-rocksdb) on with gcc 13 work?

**Comment by romanz** [comment]
Could you also try building with `clang`?

**Comment by antonilol** [comment]
`CXX='/bin/clang' cargo b` does not work, it creates similar error messages
`clang --version`:
clang version 15.0.7
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin

i tried this before going into the pain of compiling gcc11 which took an hour, maybe a different clang version will work though

`cargo b` inside a freshly cloned rust-rocksdb/rust-rocksdb (c14532d8c2a8f3f3a75ab23f90db1ec759a9f44e) works

**Comment by romanz** [comment]
It should be fixed by cherry-picking https://github.com/facebook/rocksdb/pull/11118

**Comment by romanz** [comment]
@antonilol  Please try https://github.com/romanz/electrs/pull/773/commits/a3ba928c1867cb02ab440db658b7f0f966e16e15.

**Comment by antonilol** [comment]
that works! thanks!
i also tested clang (15.0.7) and that also works

**Comment by antonilol** [comment]
this issue can be closed then when #773 is merged
